.github/workflows/gradle.yml

```diff
@@ -43,6 +64,19 @@ jobs:
         with:
           python-version: '3.7'
 
+      - name: Start local Docker registry
```

Comment:
how does this interact with the docker cache? the idea is we get access to the `Docker Caching` above? i'm not sure i full understand how these two work together.

Reply:
This uses `/tmp/docker-registry` as the data directory for a local registry. When `build_image.sh` pushes data to the local registry, this is what's being cached in `Docker Caching` (which creates a post-step at the end which saves the contents).